SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Storage/Storage.cs

```diff
@@ -224,7 +291,19 @@ await _storageAdapter.GetAsync<LogEntry>(entry => entry.Channel == channelName,
                 ProcessLogIds(channelName, batchId, idPairs);
                 logs?.AddRange(retrievedLogs);
                 return batchId;
+            });
+
+            try
+            {
+                _queue.Add(task);
             }
+            catch (InvalidOperationException)
+            {
+                throw new StorageException("The operation has been cancelled");
+            }
+
+            await FlushQueue().ConfigureAwait(false);
```

Comment:
So basically if you keep putting logs, the queue will never reach 0 and get will never return ever.